### ISSUE TITLE
Handle nested copybooks?

### ISSUE THREAD

**Issue by richardm90** [codefori/vscode-rpgle#284]
I make heavy use of nested RPG copybooks and wondered if there is an appetite to handle this within VS Code?

I miss this mostly in the peek references and hover text but would also like to see it added to the linter.

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.6.2|
|Visual Studio Code version|1.85.1|
|Operating System|linux_x64|

<details>
<summary>Active extensions</summary>

```
COBOL (cobol): 9.12.25
Code Coverage for IBM i (code-coverage-ibmi): 0.1.7
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.4.0
Db2 for IBM i (vscode-db2i): 0.6.0
ESLint (vscode-eslint): 2.4.2
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.16.0
Extension Authoring (extension-editing): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Authentication (github-authentication): 0.0.2
Grunt support for VS Code (grunt): 1.0.0
Gulp support for VSCode (gulp): 1.0.0
IBM i Debug (ibmidebug): 1.0.0
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
JSON Language Features (json-language-features): 1.0.0
Jake support for VS Code (jake): 1.0.0
JavaScript Debugger (js-debug): 1.85.0
Markdown Language Features (markdown-language-features): 1.0.0
Markdown Math (markdown-math): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
NPM support for VS Code (npm): 1.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
RPGLE (vscode-rpgle): 0.24.0
Server Ready Action (debug-server-ready): 1.0.0
Simple Browser (simple-browser): 1.0.0
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R5M0|
|Tech Refresh|3|
|CCSID|1146|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin||/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|
|-|-|-|-|-|
|bash|attr|ILEDITOR.GETNEWLIBL|QZDFMDB2.PGM|startDebugService.sh|
|chsh|iconv||||
|git|setccsid||||
|grep|||||
|ls|||||
|md5sum|||||
|sort|||||
|stat|||||
|tar|||||
</details>

**Comment by worksofliam** [comment]
In theory it should be possible, but I have not done it this far. I will stick it on my 2024 list.

**Comment by worksofliam** [comment]
I actually think I just did this with 0.27.0!

**Comment by richardm90** [comment]
I was really excited when I saw the comment so I upgraded to `0.27.0` and created a simple example but unfortunately it's not working for me.

Here's what I tried ...

1. Source file called `BASE_D.rpgleinc`

```
**free

dcl-s NAME_t varchar(40) template;
dcl-s SHORT_NAME_t varchar(20) template;
```

2. Source file called `EXAMPLE_D.rpgleinc` - note that this references `BASE_D.rpgleinc`

```
**free

/include 'BASE_D.rpgleinc'

dcl-s CUSTOMER_NAME_t like(NAME_t) template;
dcl-s CONTACT_NAME_t varchar(20) template;
```

3. Source file called `EXAMPLE1.pgm.rpgle` - note that this references `EXAMPLE_D.rpgleinc` but not `BASE_D.rpgleinc`

```
**free

ctl-opt dftactgrp(*new);

/include 'EXAMPLE_D.rpgleinc'

dcl-s contact_name like(CONTACT_NAME_t);
dcl-s customer_name like(CUSTOMER_NAME_t);
dcl-s work_name like(NAME_t);

work_name = 'Richard';
snd-msg work_name;

return;
```

If I hover over `CONTACT_NAME_t` or `CUSTOMER_NAME_t` I get information about the field but if I hover over `NAME_t` I don't see anything.

**Comment by worksofliam** [comment]
Just tested. Looks like we do, but maybe not based on path. Though that might be a separate issue from this.

<img width="1408" alt="image" src="https://github.com/user-attachments/assets/aeeccfe3-4f10-451b-8b60-cebf019b47d8">

**Comment by richardm90** [comment]
@worksofliam, OK, I think I'm getting it now.

In my example I had all three files in the same local directory though that was a sub-directory of my main project directory.

I tried the same example with my three files in the main project directory and I got the results I expected.

I then made a minor tweak to my include statements to point to the current directory (e.g. `/include './BASE_D.rpgleinc'`) and it worked with my files in a sub-directory.

1. Source file called BASE_D.rpgleinc

```
**free

dcl-s NAME_t varchar(40) template;
dcl-s SHORT_NAME_t varchar(20) template;
```
2. Source file called EXAMPLE_D.rpgleinc - note that this references BASE_D.rpgleinc

```
**free

/include './BASE_D.rpgleinc'

dcl-s CUSTOMER_NAME_t like(NAME_t) template;
dcl-s CONTACT_NAME_t varchar(20) template;
```
3. Source file called EXAMPLE1.pgm.rpgle - note that this references EXAMPLE_D.rpgleinc but not BASE_D.rpgleinc

```
**free

ctl-opt dftactgrp(*new);

/include './EXAMPLE_D.rpgleinc'

dcl-s contact_name like(CONTACT_NAME_t);
dcl-s customer_name like(CUSTOMER_NAME_t);
dcl-s work_name like(NAME_t);

work_name = 'Richard';
snd-msg work_name;

return;
```

What confused me was that it found references in my first copybook i.e. `EXAMPLE_D.rpgleinc` without need to specify `./EXAMPLE_D.rpgleinc` but not references in my second nested copybook i.e. `BASE_D.rpgleinc`. The nested copybook needed to be specified with the correct relative path.

Thank you, I'm excited again!
